validator.go

```diff
@@ -27,18 +27,18 @@ func (v *Validator) HandleChange(kind ChangeKind, p string, fi os.FileInfo, err
 	if v.parentDirs == nil {
 		v.parentDirs = make([]parent, 1, 10)
 	}
-	if p != filepath.Clean(p) {
+	if p != path.Clean(p) {
```

Comment:
I think we somehow need to make sure that there are no `\` in this path (at least for wcow). Otherwise we do the validation only based on `/` but then in diskwriter these are converted to local paths in `FromSlash()` and and the previous `\` become active.

Eg. lets say remote party sends:

```
dir foo
symlink foo/sym -> /
dir file foo\sym
file foo\sym/bar
```

I think that would pass validator atm. but would result in breakout in diskwriter.

Reply:
`sym\bar` is a valid filename in linux but I guess can't be in windows. So if validator/diskwriter run on windows then the paths should not contain `\` and it should cause a validator error if there is such path. In Linux, if there is `\` then it should be just written out verbose without making a directory separator.

If there is a backwards compatibility issue then it would also be possible that in wcow we convert all `\` to `/` before we run the validator.

Reply:
Added a check for this case now - the check is actually quite simple - we just convert from and to the slash format, and if there's a difference, the path can't be represented on the target.

This detects the path "lossiness".

---

Note: unlike I mentioned above, I don't think the reverse is an issue, since `/` isn't a valid filename character in windows? So we would never be in a scenario of trying to send a file `foo/bar\baz` from windows to linux (so the reverse issue doesn't apply).